generators/csharp/model/src/enum/StringEnumGenerator.ts

```diff
@@ -0,0 +1,317 @@
+import { csharp, CSharpFile, FileGenerator } from "@fern-api/csharp-codegen";
+import { join, RelativeFilePath } from "@fern-api/fs-utils";
+import { EnumTypeDeclaration, NameAndWireValue, TypeDeclaration } from "@fern-fern/ir-sdk/api";
+import { ModelCustomConfigSchema } from "../ModelCustomConfig";
+import { ModelGeneratorContext } from "../ModelGeneratorContext";
+
+export class StringEnumGenerator extends FileGenerator<CSharpFile, ModelCustomConfigSchema, ModelGeneratorContext> {
+    private readonly classReference: csharp.ClassReference;
+
+    constructor(
+        context: ModelGeneratorContext,
+        private readonly typeDeclaration: TypeDeclaration,
+        private readonly enumDeclaration: EnumTypeDeclaration
+    ) {
+        super(context);
+        this.classReference = this.context.csharpTypeMapper.convertToClassReference(this.typeDeclaration.name);
+    }
+
+    protected doGenerate(): CSharpFile {
+        const serializerAnnotation = csharp.annotation({
+            reference: csharp.classReference({
+                name: "JsonConverter",
+                namespace: "System.Text.Json.Serialization"
+            }),
+            argument: csharp.codeblock((writer) => {
+                writer.write("typeof(");
+                writer.writeNode(csharp.classReference(this.context.getStringEnumSerializerClassReference()));
+                writer.write("<");
+                writer.writeNode(this.classReference);
+                writer.write(">");
+                writer.write(")");
+            })
+        });
+
+        const stringEnum = csharp.class_({
+            ...this.classReference,
+            interfaceReferences: [
+                csharp.classReference({
+                    name: "IStringEnum",
+                    namespace: this.context.getCoreNamespace()
+                }),
+                csharp.classReference({
+                    name: "IEquatable",
+                    generics: [csharp.Type.reference(this.classReference)],
+                    namespace: "System"
+                })
+            ],
+            annotations: [serializerAnnotation],
+            access: "public",
+            type: "struct",
+            readonly: true
+        });
+
+        stringEnum.addConstructor({
+            access: "public",
+            parameters: [
+                csharp.parameter({
+                    name: "value",
+                    type: csharp.Type.string(),
+                    docs: "The string value of the enum."
+                })
+            ],
+            body: csharp.codeblock((writer) => {
+                writer.write("Value = value;");
```

Comment:
super nit: we try not to write semicolons directly and prefer `writer.write...Statement(...)`, in this case `writer.writerTextStatement(...)`